irohad/maintenance/metrics.cpp

```diff
@@ -9,88 +9,144 @@
+
+  auto &peers_number_gauge =
+      BuildGauge()
+          .Name("peers_number")
+          .Help("Total number peers to send transactions and request proposals")
+          .Register(*registry_);
+  auto &number_of_peers = peers_number_gauge.Add({});
+  number_of_peers.Set(storage_->getWsvQuery()->getPeers()->size());
+
+  auto &domains_number_gauge = BuildGauge()
+                                   .Name("number_of_domains")
+                                   .Help("Total number of domains in WSV")
+                                   .Register(*registry_);
+  auto &domains_number = domains_number_gauge.Add({});
+  domains_number.Set(storage_->getWsvQuery()->countDomains().assumeValue());
+
+  auto &total_number_of_transactions_gauge =
+      BuildGauge()
+          .Name("total_number_of_transactions")
+          .Help("Total number of transactions in blockchain")
+          .Register(*registry_);
+  auto &total_number_of_transactions =
+      total_number_of_transactions_gauge.Add({});
+  total_number_of_transactions.Set(
+      storage_->getWsvQuery()->countTransactions().assumeValue());
+
+  auto &number_of_signatures_in_last_block_gauge =
+      BuildGauge()
+          .Name("number_of_signatures_in_last_block")
+          .Help("Number of signatures in last block")
+          .Register(*registry_);
+  auto &number_of_signatures_in_last_block =
+      number_of_signatures_in_last_block_gauge.Add({});
+
+  auto calc_diffs = [](shared_model::interface::Block const &block) {
+    int domains_diff = 0, peers_diff = 0;
+    using namespace shared_model::interface;
+    for (Transaction const &trx : block.transactions()) {
+      for (Command const &cmd : trx.commands()) {
+        domains_diff += cmd.is<CreateDomain>() ? 1 : 0;
+        peers_diff += cmd.is<AddPeer>() ? 1 : 0;
+        peers_diff -= cmd.is<RemovePeer>() ? 1 : 0;
+      }
     }
-  }).detach();
+    return std::tuple{domains_diff, peers_diff};
+  };
 
-  return {registry};
+  block_subscriber_ = SubscriberCreator<bool, BlockPtr>::template create<
+      EventTypes::kOnBlock,
+      SubscriptionEngineHandlers::kMetrics>(
+      [&, registry = this->registry_]  /// Values are stored in registry_, hold
+                                       /// strong reference to it here
+      (auto &, BlockPtr pblock) {
+        // block_height is captured by reference because it is stored inside
+        // registry_, which is shared_ptr
+        assert(pblock);
+        block_height.Set(pblock->height());
+        number_of_signatures_in_last_block.Set(
+            boost::size(pblock->signatures()));
+        total_number_of_transactions.Increment(
+            boost::size(pblock->transactions()));
+        auto [domains_diff, peers_diff] = calc_diffs(*pblock);
+        number_of_peers.Increment(peers_diff);
+#if 1
+        domains_number.Increment(domains_diff);
+#else  // no need to querry DB but here is a way to do
+        if (domains_diff) {
```

Comment:
Maybe store this in a GitHub issue instead to avoid dead code?

Reply:
removed